src/main/java/gregtech/api/items/toolitem/IGTTool.java

```diff
@@ -173,11 +174,14 @@ default ItemStack get(Material material) {
         }
 
         // Set tool and material enchantments
-        Object2IntMap<Enchantment> enchantments = toolProperty.getEnchantments();
+        Object2IntMap<Enchantment> enchantments = new Object2IntOpenHashMap<>();
+        enchantments.putAll(toolProperty.getEnchantments());
         enchantments.putAll(toolStats.getDefaultEnchantments(stack));
         enchantments.forEach((enchantment, level) -> {
             if (stack.getItem().canApplyAtEnchantingTable(stack, enchantment)) {
-                stack.addEnchantment(enchantment, level);
+                int levelIncrease = toolStats.getDefaultEnchantmentLevelIncrease(stack).getOrDefault(enchantment, 0)
+                        * toolProperty.getToolHarvestLevel();
+                stack.addEnchantment(enchantment, level + levelIncrease);
```

Comment:
I think we need to manually check min/max levels of enchantment here, otherwise we would get things like underleveled/overleveled enchantments, and enchantments with level exceeding 127 which minecraft doesn't seem to appreciate that much. (hard cap might be `Short.MAX_VALUE`, the only place `Byte.MAX_VALUE` is enforced is `ItemStack#addEnchantment` and elsewhere I just see shorts being used. #Mojang #MCCode)

Actually uh overleveled enchantments sounds hype?? Maybe a feature??? If you want to implement `EnchantmentLevel` stuff I suggested above, you may be able to shove in extra bool field to allow/disallow overenchanting. Or just receiving maximum enchant level would be better. idk.